### DISCUSSION TITLE
Unique Index to replace id

### DISCUSSION THREAD

**Discussion by Faaatman** [isar/isar#285] [Help] [answered]
I use UUIDs for chat messages in my app and backend.
What I'm trying to achieve is I cache the messages that are being sent locally where Isar assigns an id for it and then send it to the backend so the chat is seamless and fast.
The problem is sometimes the message's data changes (ex: createdAt which differs from when it was created locally and when it was created in the backend). The solution that now works for me is using the UUID as an index and query based on it, if id's match then put normally if they don't then delete the one local and use the one from the backend. this solution however breaks putAll when retrieving a bunch of messages and I have to query for every single message I get.
Is there a way where the collection has a unique index to be treated as the id?

**Comment by Faaatman** [comment, marked as answer]
I used `replaceOnConflict` flag which solves my issue
ex: `isar.collectionName.putAll(items, replaceOnConflict: true)`

**Comment by PapyElGringo** [comment]
@Faaatman How did you used UID as index since it's must be int? I'm facing the same scenario

**Reply by Faaatman** [reply]
I kept the id the same and added the uuid as follows:
```dart
@Index(unique: true)
final String uuid;
```
after doing that I can use the uuid when searching, inserting and, deleting.
don't forget to set `replaceOnConflict` to `true`
